allennlp/interpret/influence_interpreters/influence_interpreter.py

```diff
@@ -0,0 +1,361 @@
+from os import PathLike
+import re
+from typing import List, Optional, NamedTuple, Sequence, Union, Dict, Any
+
+import torch
+from torch import autograd
+
+from allennlp.common import Registrable, Lazy, plugins
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import int_to_device
+from allennlp.data import Instance, DatasetReader, DatasetReaderInput, Batch
+from allennlp.data.data_loaders import DataLoader, SimpleDataLoader
+from allennlp.models import Model, Archive, load_archive
+from allennlp.nn.util import move_to_device
+
+
+class TrainInstanceInfluence(NamedTuple):
```

Comment:
Not sure that `Train` really adds much to this name.  `InstanceInfluence` is likely sufficient, and already implies that it's a training instance.